### PR TITLE
Add budget feature

### PR DESCRIPTION
**Pull request by choonkit-nus**

### CODE REVIEW

File: src/main/java/seedu/nuscents/data/TransactionList.java [hunk 1]
```diff
@@ -79,8 +80,37 @@ public void filterTransaction(TransactionCategory category) {
...
+        float budget = getBudget();
+        float expense = getTotalExpense();
+        if (expense > budget) {
+            isWithinBudget = false;
+        } else {
+            isWithinBudget = true;
+        }
+        return isWithinBudget;
+    }
+}
```
Comment (spaceman03) [inline src/main/java/seedu/nuscents/data/TransactionList.java:116]: Perhaps you could add a newline between the 2 methods

Comment (choonkit-nus) [reply src/main/java/seedu/nuscents/data/TransactionList.java:116]: Oh ok, will edit it to make it more readable!

File: src/main/java/seedu/nuscents/storage/Storage.java [hunk 2]
```diff
@@ -221,5 +221,31 @@ private byte[] readFile(String filePath) throws IOException {
...
+            }
+            sc.close();
+        } catch (FileNotFoundException e) {
+            Ui.showReadDataError();
+            File file = new File("./data/budget.txt");
+            file.getParentFile().mkdirs();
+            file.createNewFile();
+        }
+        return "0";
+    }
```
Comment (spaceman03) [inline src/main/java/seedu/nuscents/storage/Storage.java:249]: Nice to see that you have created a separate file to store the budget. 😃

**Comment by lckjosh** [comment]
LGTM!

**Pull request merged**